packages/next-app/src/app/internalmanager/(components)/TokenSelect.tsx

```diff
@@ -99,49 +94,62 @@ function TokenModal({
     userAddress: addressLower as `0x${string}`,
   });
 
-  useEffect(() => {
-    moralisGetTokens({
-      userAddress: addressLower,
-      chain: chain!.name as "Goerli",
-    }).then((res) => {
+  const internalBalancesTokenAdresses = internalBalanceData?.user
+    ?.userInternalBalances
+    ? internalBalanceData.user.userInternalBalances.map(
+        (token) => token.tokenInfo.address
+      )
+    : [];
+
+  async function getWalletBalance(
+    internalBalancesTokenAdresses: `0x${string}`[]
+  ) {
+    const walletBalanceData: TokenWalletBalance[] = [];
+    const walletBalancePromises = internalBalancesTokenAdresses.map(
+      async (tokenAddress) => {
+        const tokenData = await fetchBalance({
```

Comment:
https://linear.app/bleu-llc/issue/BAL-363/substitute-the-use-of-preparewritecontract-for-wagmi-hooks

Reply:
not sure I understand the issues linked here, do you want to substitute `fetchBalance`?

Reply:
I'm using a wagmi function, but I want to try to use wagmi Hooks instead :P